dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/QueryUtils.java

```diff
@@ -54,6 +53,9 @@
  * @author Morten Olav Hansen <mortenoh@gmail.com>
  */
 public final class QueryUtils {
+
+  private static final String UNABLE_TO_PARSE = "Unable to parse `";
```

Comment:
This is bordering on making the code less readable. It is good to use constants for "magic numbers" and similar working strings but when it comes to error messages, fields names and such repetition isn't necessarily the issue. We tend to keep inline strings for such as it helps to see what the code is doing or producing without having to jump somewhere else and then put it together in you mind. 
I think in this instance it is fine to use a constant just wanted to give you the gist of how we mostly do things even if tools might warn about duplicates.

Reply:
could easily argue that it's more readable & reduces mental processing required
- with the constant, hovering over it will show its value (no need to jump anywhere)
- because it's used in many other spots, and because you now know the value, you don't now have to manually read each string to make sure they all match.
- without the constant, you have to read every character in every string to know it's value

Reply:
I meant to click approve :D Yes in this case constant work. In cases where json is composed or multiple string parts are combined, like SQL building or complex error message building you cannot hover all at the same time, and even if you would need you mental capacity to substitute the constants with their values instead of using it to reason about the entire value put together.